locksmith/src/utils/promises.ts

```diff
@@ -0,0 +1,18 @@
+export const waitOrTimeout = async (
+  promise: Promise<any>,
+  delay: number,
+  responseIfTimeout: () => any
```

Comment:
Do these args really need to be passed? `delay` could have directly `TIMEOUT_ON_SUBGRAPH` set as default value, then just a smaller error message string